io/net/bonding.py

```diff
@@ -311,67 +301,68 @@ def bond_setup(self, arg1, arg2):
             if process.system(peer_cmd, shell=True, ignore_status=True) != 0:
                 self.fail("bond setup command failed in peer machine")
 
-    def test_bonding(self):
+    def test_setup(self):
         '''
         bonding the interfaces
         work for multiple interfaces on both host and peer
         '''
-        self.log.info("Bonding")
-        msg = "[ -d /sys/class/net/%s ]" % self.bond_name
+        msg = "[ -d %s ]" % self.bond_dir
         cmd = "ssh %s@%s %s" % (self.user, self.peer_first_ipinterface, msg)
         if process.system(cmd, shell=True, ignore_status=True) == 0:
             self.fail("bond name already exists on peer machine")
         if os.path.isdir(self.bond_dir):
             self.fail("bond name already exists on local machine")
-        self.log.info("TESTING FOR MODE %s", self.mode)
-        self.log.info("-------------------------------------------------")
         if self.peer_bond_needed:
             self.bond_setup("peer", "")
         self.bond_setup("local", self.mode)
-        process.run(self.bond_status, shell=True, verbose=True)
+        self.log.info(genio.read_file(self.bond_status))
         self.ping_check()
+        self.error_check()
+
+    def test_run(self):
         self.bond_fail(self.mode)
         self.log.info("Mode %s OK", self.mode)
+        self.error_check()
 
-    def tearDown(self):
+    def test_cleanup(self):
```

Comment:
why this i think `tearDown` was good choice

Reply:
@PraveenPenguin Agreed, but wanted bond remove as a seperate test.
Suggestions on handling are welcome..
Or give me a day to get a better flow :)

Reply:
sure as offline discussion we take this as enhancement , merging for now

Reply:
@Sure, thanks.